plugins/modules/monitoring/alerta_customer.py

```diff
@@ -0,0 +1,194 @@
+        self.headers = {"Content-Type": "application/json"}
+
+        if module.params.get('api_key', None):
+            self.headers["Authorization"] = "Key %s" % module.params['api_key']
+        else:
+            self.headers["Authorization"] = basic_auth_header(module.params['api_username'], module.params['api_password'])
+
+    def send_request(self, url, data=None, method="GET"):
+        response, info = fetch_url(self.module, url, data=data, headers=self.headers, method=method)
+
+        status_code = info["status"]
+        if status_code == 401:
+            self.module.fail_json(failed=True, response=info, msg="Unauthorized to request '%s' on '%s'" % (method, url))
+        elif status_code == 403:
+            self.module.fail_json(failed=True, response=info, msg="Permission Denied for '%s' on '%s'" % (method, url))
+        elif status_code == 404:
+            self.module.fail_json(failed=True, response=info, msg="Not found for request '%s' on '%s'" % (method, url))
+        elif status_code == 200 or status_code == 201:
+            return self.module.from_json(response.read())
+        self.module.fail_json(failed=True, response=info, msg="Alerta API error with HTTP %d for %s" % (status_code, url))
+
+    def get_customers(self):
+        url = "%s/api/customers" % (self.alerta_url)
+        response = self.send_request(url)
+        return response
+
+    def create_customers(self):
+        url = "%s/api/customer" % (self.alerta_url)
+
+        payload = {
+            'customer': self.customer,
+            'match': self.match,
+        }
+
+        payload = self.module.jsonify(payload)
+        response = self.send_request(url, payload, 'POST')
+        return response
+
+    def delete_customers(self, id):
+        url = "%s/api/customer/%s" % (self.alerta_url, id)
+
+        response = self.send_request(url, None, 'DELETE')
+        return response
+
+    def check_customer(self, customer):
+        for i in customer['customers']:
+            if self.customer == i['customer'] and self.match == i['match']:
+                return i['id']
+        return None
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(choices=['present', 'absent'], default='present'),
+            customer=dict(type='str', required=True),
+            match=dict(type='str', required=True),
+            alerta_url=dict(type='str', required=True),
+            api_username=dict(type='str'),
+            api_password=dict(type='str', no_log=True),
+            api_key=dict(type='str', no_log=True),
+        ),
+        required_together=[['api_username', 'api_password']],
+        mutually_exclusive=[['api_username', 'api_key']],
+        supports_check_mode=True
+    )
+
+    alerta_iface = AlertaInterface(module)
+
+    if alerta_iface.state == 'present':
+        response = alerta_iface.get_customers()
+        if alerta_iface.check_customer(response):
```

Comment:
Do I understand it correctly that both `name` and `match` must be a precise match for a customer? I.e. two different customers can have the same `name` (but different `match`), or the same `match` (but different `name`)?

Reply:
Yes, correct. The `match` is the corresponding username and `name` the name of a customer (view). One user can have more than one customers, same for customers.

Example:

- User `developer` with customer view `dev` and `billing`
- Customer view `dev` for user `developer` and `admin`